### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by simicmarin**
## Changes made:

- Gradle wrapper updated to 8.0.2
- AGP updated to 8.0.0
- Kotlin updated to 1.9.21
- Version catalog used

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[versions]
+gradle = "8.0.0"
```
Comment (JonatanPlesko) [inline gradle/libs.versions.toml]: Maybe `android-gradle-plugin` is a bit more accurate, Gradle version is a different thing. Also, you can use 8.2.0 here as it's the latest version.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+kotlin = "1.9.21"
```
Comment (JonatanPlesko) [inline gradle/libs.versions.toml]: 1.9.22 is out

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Fri Apr 17 09:36:20 CEST 2020
+#Sun Dec 31 11:49:46 CET 2023
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.2-bin.zip
```
Comment (JonatanPlesko) [inline gradle/wrapper/gradle-wrapper.properties]: Why not 8.5?

**Review by JonatanPlesko — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (simicmarin) [reply gradle/libs.versions.toml]: Agreed 👍

**Pull request merged**